cpp/src/copying/get_element.cu

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/copying.hpp>
+#include <cudf/scalar/scalar_factories.hpp>
+#include <cudf/scalar/scalar_device_view.cuh>
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/dictionary/dictionary_column_view.hpp>
+
+namespace cudf {
+namespace experimental {
+namespace detail {
+
+namespace {
+
+struct get_element_functor {
+  
+  template<typename T>
+  std::enable_if_t<is_fixed_width<T>(), 
+  std::unique_ptr<scalar>> operator()(
+    column_view const& input,
+    size_type index,
+    cudaStream_t stream = 0,
+    rmm::mr::device_memory_resource *mr = rmm::mr::get_default_resource())
+  {
+    auto s = make_fixed_width_scalar(data_type(type_to_id<T>()), stream, mr);
+
+    using ScalarType = cudf::experimental::scalar_type_t<T>;
+    auto typed_s = static_cast<ScalarType *>(s.get());
+    
+    auto device_s = get_scalar_device_view(*typed_s);
+    auto device_col = column_device_view::create(input, stream);
+    
+    thrust::for_each_n(rmm::exec_policy(stream)->on(stream), 
```

Comment:
This seems overcomplicated. Why not use a cudaMemcpyAsync  from the buffer pointer of the column view to the buffer pointer of the scalar? If this were a method on `column_view` that would be more natural.

Reply:
`d_col.is_valid(index)` needs compute. So, kernel needs to be launched anyway.

Reply:
if it is `column_view` member method, should it return `T` or `scalar`?